_rules/SC2-4-6-descriptive-labels.md

```diff
@@ -148,23 +199,58 @@ Label is included in accessibility tree, but not visible, and does not describe
 <input aria-labelledby="label_fname" type="text" name="fname"/>
 ```
 
+#### Failed Example 6
+
+Accessible name created through "aria-label" does not describes the purpose of the associated element.
+
+```html
+<input aria-label="Menu" id="fname" type="text" name="fname"/>
+```
+
+#### Failed Example 7
+
+The `label` in itself does not describe the purpose of the form field, and while the context can be relied upon to differentiate form fields from each other, it is not sufficient to rely on the context for describing the entire purpose of the form field.
```

Comment:
Just a suggestion of rewording. In the description of the example you emphasized on the fact that the context is not sufficient for describing the purpose of the form field. But if I understand well your example, the problem is, in the first place, that the label does not describe the purpose at all, and so, the context can't applied on a label that is not descriptive.  

If it's right :-) I suggest: 

> The `label` in itself does not describe the purpose of the form field. While the context could have been relied upon to differentiate form fields from each other, the fact that the label is not descriptive makes the use of this context ineffective.